pkg/render/common/elasticsearch/service.go

```diff
@@ -25,13 +25,26 @@ const (
 	linseedEndpoint     = "https://tigera-linseed.%s.svc"
 	httpsFQDNEndpoint   = "https://tigera-secure-es-gateway-http.%s.svc.%s:9200"
 	linseedFQDNEndpoint = "https://tigera-linseed.%s.svc.%s"
+
+	// Handle for multi tenant
+	guardianEndpoint       = "https://guardian.calico-system.svc"
+	guardinaFQDNEndpoint   = "https://guardian.calico-system.svc.%s"
+	fluentdLinseedEndpoint = "https://tigera-linseed"
 )
 
-func LinseedEndpoint(osType rmeta.OSType, clusterDomain, namespace string) string {
-	if osType == rmeta.OSTypeWindows {
+func LinseedEndpoint(osType rmeta.OSType, clusterDomain, namespace string, isManagedCluster bool, isFluentd bool) string {
+	switch {
+	case isManagedCluster && isFluentd:
+		return fluentdLinseedEndpoint
```

Comment:
We should have a comment explaining why this is an exception, otherwise this code is pretty opaque to anyone trying to reverse engineer it.